GiniVision/Classes/ReviewViewController.swift

```diff
@@ -62,35 +76,41 @@ public typealias ReviewErrorBlock = (_ error: ReviewError) -> ()
     fileprivate var imageViewTopConstraint: NSLayoutConstraint!
     fileprivate var imageViewTrailingConstraint: NSLayoutConstraint!
     fileprivate var metaInformationManager: ImageMetaInformationManager?
+    fileprivate var currentDocument:GiniVisionDocument?
     
     // Images
     fileprivate var rotateButtonImage: UIImage? {
         return UIImageNamedPreferred(named: "reviewRotateButton")
     }
     
     // Output
-    fileprivate var successBlock: ReviewSuccessBlock?
-    fileprivate var errorBlock: ReviewErrorBlock?
+    fileprivate var successBlock: ReviewScreenSuccessBlock?
+    fileprivate var failureBlock: ReviewScreenFailureBlock?
+    
     
     /**
-     Designated intitializer for the `ReviewViewController` which allows to set a success block and an error block which will be executed accordingly.
-     
+     Designated initializer for the `ReviewViewController` which allows to set a success block and an error block which will be executed accordingly.
      
-     - parameter imageData: JPEG representation as a result from the camera or camera roll.
+     - parameter document:  JPEG representation or PDF as a result from the camera, camera roll or file explorer.
      - parameter success:   Success block to be executed when image was rotated.
      - parameter failure:   Error block to be executed if an error occured.
      
-     - returns: A view controller instance allowing the user to review a picture of a document.
+     - returns: A view controller instance allowing the user to review a picture.
      */
-    public init(_ imageData: Data, success: @escaping ReviewSuccessBlock, failure: @escaping ReviewErrorBlock) {
+    public init(_ document: GiniVisionDocument, successBlock: @escaping ReviewScreenSuccessBlock, failureBlock: @escaping ReviewScreenFailureBlock) {
         super.init(nibName: nil, bundle: nil)
         
         // Set callback
-        successBlock = success
-        errorBlock = failure
+        self.successBlock = successBlock
+        self.failureBlock = failureBlock
+        
+        // Initialize currentDocument
+        currentDocument = document
         
         // Set meta information manager
-        metaInformationManager = ImageMetaInformationManager(imageData: imageData)
+        if document.type == .Image {
```

Comment:
Having a "Document" metaphor is a cool idea. However, I'm still seeing these type checks all over the place. This defeats the purpose of hiding PDFs and images in a Document. I think you should rethink some of the architecture to make ti possible to not care what kind of document it is in places such as `ReviewViewController`.

Reply:
Actually I wanted to move the _ImageMetaInformationManager_ to the GiniImageDocument class, but since I have other things to do, I was waiting to this PR to be approved.